subgraphs/abyssmal/src/index.ts

```diff
@@ -1,15 +1,70 @@
-import { Approvals } from "../generated/schema";
-import { Approval } from "../generated/Abyssmal/Abyssmal";
+import {
+  ApprovalTx,
+  ApprovalForAllTx,
+  MintCostUpdatedTx,
+  NFTMintedTx,
+  OwnershipTransferredTx,
+  TransferTx,
+} from "../generated/schema";
+import {
+  Approval,
+  ApprovalForAll,
+  MintCostUpdated,
+  NFTMinted,
+  OwnershipTransferred,
+  Transfer,
+} from "../generated/Abyssmal/Abyssmal";
+import { BigInt } from "@graphprotocol/graph-ts";
+import { createId } from "../../../helpers/utils";
 
 export function handleApproval(event: Approval): void {
-  const logIndex = event.logIndex;
-  const transactionHash = event.transaction.hash;
+  const id = createId(event.transaction.hash, event.logIndex);
+  const approvalTx = new ApprovalTx(id);
+  approvalTx.owner = event.params.owner.toHex();
+  approvalTx.approved = event.params.approved.toHex();
+  approvalTx.tokenId = event.params.tokenId;
+  approvalTx.save();
+}
+
+export function handleApprovalForAll(event: ApprovalForAll): void {
+  const id = createId(event.transaction.hash, event.logIndex);
+  const approvalForAllTx = new ApprovalForAllTx(id);
+  approvalForAllTx.owner = event.params.owner.toHex();
+  approvalForAllTx.operator = event.params.operator.toHex();
+  approvalForAllTx.approved = event.params.approved;
+  approvalForAllTx.save();
+}
+
+export function handleMintCostUpdated(event: MintCostUpdated): void {
+  const id = createId(event.transaction.hash, event.logIndex);
+  const mintCostUpdatedTx = new MintCostUpdatedTx(id);
+  mintCostUpdatedTx.oldMintCost = event.params.oldMintCost;
+  mintCostUpdatedTx.newMintCost = event.params.newMintCost;
+  mintCostUpdatedTx.save();
+}
+
+export function handleNFTMinted(event: NFTMinted): void {
+  const id = createId(event.transaction.hash, event.logIndex);
+  const nftMintedTx = new NFTMintedTx(id);
+  nftMintedTx.mintedBy = event.params.mintedBy.toHex();
```

Comment:
What is the difference between `toHex()` and `toHexString()` here?

Reply:
Hex as 0x00 or hex as "0x00"